src/response.rs

```diff
@@ -309,6 +309,45 @@ mod tests {
         assert_eq!(parsed_auth, expected_auth);
     }
 
+    #[test]
+    fn parse_denied_authorization() {
+        let xml_response = r##"
+        <?xml version="1.0" encoding="UTF-8"?>
+        <status>
+          <authorized>false</authorized>
+          <reason>application key is missing</reason>
+          <plan>sample</plan>
+          <usage_reports>
```

Comment:
Why is this returning reports when not authorized? :thinking:

Reply:
Because when not authorized you still want to know which metrics weren't authorized and when the periods end.

Reply:
Right, but the reason shouldn't be `application key is missing`. It should be the message about passed limits.

Reply:
But the limits were not exceeded in this particular case, just the caller did not specify an app_key and that's why it is not authorized (but still giving you the usage report).

Reply:
Right, I missed that case :+1: